src/workerd/api/html-rewriter.c++

```diff
@@ -446,7 +447,10 @@ kj::Promise<void> Rewriter::write(
 
 kj::Promise<void> Rewriter::end() {
   KJ_ASSERT(maybeWaitScope == nullptr);
-  return getFiberPool().startFiber([this](kj::WaitScope& scope) {
+  // We use kj::coCapture here to ensure that the `this` capture is maintained
+  // appropriately after the co_await finishWrite()
+  return getFiberPool().startFiber(
+      kj::coCapture([this](kj::WaitScope& scope) -> kj::Promise<void> {
```

Comment:
Whoa... Here we're mixing fibers with coroutines. I think this is confusing. Only stuff up until the first `co_await` will actually run in the fiber, presumably. The `WaitScope` is only valid in the fiber.

IMO we shouldn't mix these. Let's make a rule that functions which take `kj::WaitScope&` as an input must _not_ be coroutines. We can even codify this by having the coroutine adapter fail to compile if one of the parameters is `kj::WaitScope`.